activerecord/lib/active_record/migration.rb

```diff
@@ -1286,6 +1295,7 @@ def invalid_target?
       def execute_migration_in_transaction(migration, direction)
         return if down? && !migrated.include?(migration.version.to_i)
         return if up?   &&  migrated.include?(migration.version.to_i)
+        return if migration.skip
```

Comment:
Why should a bit of DSL be added if this is all what it actually does? You already figured out how to accomplish what you needed: an early return/guard clause in your migration.

Reply:
Adding the early return/guard it prevents to run the migration but when I remove the return/guard it won't try to run it again 😅